discord_thread_summarizer_using_openai/main.bal

```diff
@@ -0,0 +1,75 @@
+import ballerina/io;
+import ballerina/log;
+import ballerina/time;
+
+const TYPE_FORUM_CHANEL = 15;
+const time:Seconds ONE_DAY_IN_SECONDS = 86400;
+configurable int forumChannelId = ?;
+
+function readThreads(ActiveThreads activeThreads) returns error? {
+
+    foreach ChannelThread thread in activeThreads.threads {
+
+        // check if the thread has any recent messages
+        if !check hasRecentMessages(thread) {
+            continue;
+        }
+
+        string prompt = check constructPrompt(thread);
+        io:println(prompt);
+
+        string|error generateChatCompletionResult = generateChatCompletion(prompt);
+
+        if generateChatCompletionResult is string {
+            io:println(generateChatCompletionResult);
+        } else {
+            log:printError("Error occured when connecting to OpenAI: ", generateChatCompletionResult, stackTrace = generateChatCompletionResult.stackTrace());
+        }
+        io:println("--------------------------------------------------\n\n");
+    }
+}
+
+function constructPrompt(ChannelThread thread) returns string|error {
+    string prompt = string `${PROMPT}Thread URL: ${getThreadURL(thread.guild_id, thread.id)}${"\n"}Title: ${thread.name}${"\n"}Question: `;
+    // reverse the array so that the messages are in chronological order
+    Message[] allMessages = (check getMessages(thread.id)).reverse();
+
+    boolean firstMessage = true;
+    foreach Message message in allMessages {
+        string formattedTimestamp = message.timestamp;
+        if message.content == "" {
+            io:println(string `The message from ${message.author.username} is empty.`);
+            continue;
+        }
+        // first message is the question
```

Comment:
Doesn't add value?

Reply:
Addressed in https://github.com/ballerina-guides/ai-samples/pull/69/commits/9018e16abefb68f7d46cfbc2a86a00240ce6ea5f